### ISSUE TITLE
Proposal: Support for CRS geolocation

### ISSUE THREAD

**Issue by PerspectivesLab** [buildingSMART/BCF-API#307]
many software in the industry support BCF, cartesian information for "Point" datatype is mostly referenced relative to a 3D model, or the local cartesian space of the editor
it would be nice to introduce a "GeoPoint" that supports CRS's ex "EPSG:4978" 

```
<xs:complexType name="GeoPoint">
        <xs:element name=CRS"" type="xs:string" />
        <xs:sequence>
            <xs:element name="X" type="xs:double"/>
            <xs:element name="Y" type="xs:double"/>
            <xs:element name="Z" type="xs:double"/>
        </xs:sequence>
    </xs:complexType>
```

**Comment by GeorgDangl** [comment]
Hey @PerspectivesLab, thank you for raising this issue!

We've discussed it in one of our calls, and we think the feature sounds interesting. Do you have some more information about the use case of geo-referencing points in BCF?

As of now, BCF coordinates are always relative to the model. So we thought that if actual coordinates are required, the model could be exported in such a way. What do you think?

**Comment by PerspectivesLab** [comment]
Hi @GeorgDangl , the idea behind the GeoPoint is to be able to use them in the global space,
we are currently using BCF annotations in **augmented reality** in geo space ( EPSG ) for reviewing and surveying, **without** any reference to a 3D model ( ex: a blank field for topo )
we understood the BCF is tied to a IFC model by their id's, but also works **without** any tie.
it would be nice to have a BCF **standalone** file and display it in compatible software ( for topo in revit, google earth, etc ) like a layer, but for this there is a need to write down the reference ( local or global ) for the location values, ex for bitmap :
Bitmap => <xs:element name="Location" type="Point"/> => there is no precision if local or **global coordinates** thus the GeoPoint that indicates it in world position, and not local


 


 



<xs:element name="Location" type="Point"/>

**Comment by GeorgDangl** [comment]
Hey @PerspectivesLab, I think that issue needs a bit more input from some other implementers, so let's wait and hope to get some more comments.

**Comment by PerspectivesLab** [comment]
Hi @GeorgDangl , yes it does need indeed, this all part is about global georeferencing, that would invoke connecting to the surveying world, but i personally think that would enhance the BCF roadmap, opening a more global and other format connectivity, what do you think @Moult

**Comment by Moult** [comment]
This is an interesting usecase. There are a few possible implementation issues I can think of:

 1. This only works on IFC4 and up. IFC2X3 does not support geolocation and cannot define any CRS (apart from the fallback pset which nobody has implemented).
 2. The IFC model only defines one CRS. Should this "GeoPoint" be allowed to use a different CRS? If so, is the BCF viewing software expected to transform from the GeoPoint CRS to the project CRS before converting to the local engineering coordinates?
 3. Should the GeoPoint include all necessary CRS definitions like IFC, such as not only the Projected CRS name as an EPSG code, but also the VerticalDatum as well? Vertical reprojections (see point 2) could be complex for software to implement.

The pros of geopoint:

 1. Get BCF topics in world space, agnostic of model. This makes it easy to integrate with survey equipment.
 2. Use BCF topics across multiple models with different map conversion parameters (i.e. different false origins). (technically you could still do this without GeoPoint, but it would be a more explicit way of doing so)

Cons:

 1. Doesn't support IFC2X3.
 2. If reprojection is supported, this adds significant implementation effort and potential software dependencies.

**Comment by PerspectivesLab** [comment]
Hi @Moult thanks for your feedback, 
1/ if try to foresee the next evolutions of BCF, will they be towards IFC4 or IFC2X3 ?, i think IFC2X3 is more about a worldwide software version compatibility, i am wrong to say next BCF features should be thought towards IFC4 and not IFC2X3, as the market should follow ?
2/ yes, i think the different client software should implement their own dependency/solutions for the calculations and conversions between or with a CRS's: ex proj4
3/ the EPSG code might be enough, ex 4978 for world Cartesian data, but other EPSG's might be more complex for sure.

@GeorgDangl  any news about implementing/roadmapping/using EPSG's in BCF ? thx

**Comment by GeorgDangl** [comment]
Hey guys, we've just talked about this again in today's call. Unfortunately, we've decided to not go along with this proposal.

We see the use cases, and like the idea, but the two main points for us were:
* It's essentially already possible, since BCF topics have a link to one (or multiple) IFC files, and IFC supports geo coordinates. So sticking with the model-relative coordinates of BCF makes supporting geo coordinates already possible.
* It would add a lot of implementation effort for anyone who wants to properly import BCF data
